lib/logicalDevice.js

```diff
@@ -59,4 +63,58 @@ LogicalDevice.prototype.getVolume = function(cb) {
   cb(sum / this.devices.length);
 };
 
+/**
+ * Create a Search Instance (emits 'DeviceAvailable' with a found Logical Sonos Component)
+ * @param  {Function} Optional 'DeviceAvailable' listener (sonos)
+ * @return {Search/EventEmitter Instance}
+ */
+var search = function(callback) {
+  var search = sonosSearch();
+
+  search.once('DeviceAvailable', function(device) {
+    device.getTopology(function(err, topology) {
+      if (err) callback(err);
+      else {
+
+        var devices = topology.zones;
+        var groups = {};
+
+        // bucket devices by groupid
+        devices.forEach(function(device) {
+
+          if (device.name === 'BRIDGE' || device.name === 'BOOST') return; // devices to ignore in search
+
+          if (!groups[device.group]) groups[device.group] = { members: [] };
+
+          var parsedLocation = url.parse(device.location);
+          var sonosDevice = new Sonos(parsedLocation.hostname, parsedLocation.port);
+
+          if (device.coordinator === 'true') groups[device.group].coordinator = sonosDevice;
+          groups[device.group].members.push(sonosDevice);
+
+        });
+
+        // initialze all of the logical devices brefore callback
+        var logicalDevices = Object.keys(groups).map(function(groupId) {
+          var group = groups[groupId];
+          return new LogicalDevice(group.members, group.coordinator);
+        });
+
+        async.forEach(logicalDevices, function(device) {
+          device.initialize(function(err) {
+            if (err) callback(err);
+            else {
+              callback(null, logicalDevices);
```

Comment:
This line is called every time a device is initialized and causes the search to return a full list of devices multiple times. 

See stephen/node-sonos#1. Sorry I just saw this PR, I should've pulled into @stephen's logical-search branch